packages/server/modules/shared/helpers/typeHelper.ts

```diff
@@ -2,9 +2,29 @@ import { Express } from 'express'
 
 export type Nullable<T> = T | null
 export type Optional<T> = T | undefined
+export type MaybeNullOrUndefined<T> = T | null | undefined
 export type MaybeAsync<T> = T | Promise<T>
+export type MaybeFalsy<T> = T | null | undefined | false | '' | 0
 
 export type SpeckleModule = {
-  init: (app: Express) => MaybeAsync<void>
-  finalize: (app: Express) => MaybeAsync<void>
+  /**
+   * Initialize the module
+   * @param app The Express instance
+   * @param isInitial Whether this initialization method is being invoked for the first time in this
+   * process. In tests modules can be initialized multiple times.
+   */
+  init: (app: Express, isInitial: boolean) => MaybeAsync<void>
```

Comment:
Why are these functions MaybeAsync? Doesn't it complicate our lives?

Reply:
@gjedlicska I don't think so, if you look inside ./modules/index.js you'll see that both init() & finalize() are awaited, so they already support being async. And if init() is synchronous that will work also, the `await` will be essentially ignored.

So it appears that already before this PR these functions supported both async and non-async return types.